main/http_server/http_server.c

```diff
@@ -363,12 +364,18 @@ static esp_err_t GET_system_info(httpd_req_t * req)
         return ESP_FAIL;
     }
 
+
     char * ssid = nvs_config_get_string(NVS_CONFIG_WIFI_SSID, CONFIG_ESP_WIFI_SSID);
     char * hostname = nvs_config_get_string(NVS_CONFIG_HOSTNAME, CONFIG_LWIP_LOCAL_HOSTNAME);
+    uint8_t mac[6];
+    char * formattedMac[17]
```

Comment:
You'll need to free this towards the end of this function.
> free(formattedMac);

Reply:
Will do, thank you!

Reply:
formattedMac isn't malloc'd and is on the stack. I don't think you should (or can) Free() it.

Reply:
Hopefully @WantClue or I can test this out soon and get it merged.

Reply:
Ah yep, my bad missed that. 

Noticed that you are missing semicolon in a few places, e.g.
> char * formattedMac[17]

btw do you want the mac of the AP? If not I believe we want `WIFI_IF_STA` and not `ESP_IF_WIFI_AP`.

We probably want something like this.
```
uint8_t mac[6];
char formattedMac[18];
esp_wifi_get_mac(WIFI_IF_STA, mac);
snprintf(formattedMac, 18, "%02X:%02X:%02X:%02X:%02X:%02X", mac[0], mac[1], mac[2], mac[3], mac[4], mac[5]);
```

Reply:
Or this if you want to store the string in heap memory instead.
> char * formattedMac = (char*) malloc(18);

Reply:
I believe I want the MAC of whatever is accessing the network, so not the AP.

Reply:
> uint8_t mac[6];
> char formattedMac[18];
> esp_wifi_get_mac(WIFI_IF_STA, mac);
> snprintf(formattedMac, 18, "%02X:%02X:%02X:%02X:%02X:%02X", mac[0], mac[1], mac[2], mac[3], mac[4], mac[5]);

Stack should be fine here I think.

Reply:
Yep. I think so too.

Reply:
I compiled it with the above suggestion and looks good.
```
{
	"power":	3.369999885559082,
	"voltage":	5486.25,
	"current":	620,
	"temp":	30,
	"vrTemp":	0,
	"hashRate":	0,
	"bestDiff":	"0",
	"bestSessionDiff":	"0",
	"freeHeap":	201524,
	"coreVoltage":	1200,
	"coreVoltageActual":	1202,
	"frequency":	600,
	"ssid":	"Andersson-Safe",
	"macAddr":	"AB:CD:EF:47:4C:38",
	"hostname":	"eu-home-miner03",
	"wifiStatus":	"Connected!",
	"sharesAccepted":	0,
	"sharesRejected":	0,
	"uptimeSeconds":	9,
	"asicCount":	1,
	"smallCoreCount":	1276,
	"ASICModel":	"BM1368",
	"stratumURL":	"solo.ckpool.org",
	"stratumPort":	3333,
	"stratumUser":	"<address>",
	"version":	"v2.1.10-31-gb6a2bfe-dirty",
	"boardVersion":	"401",
	"runningPartition":	"factory",
	"flipscreen":	1,
	"overheat_mode":	0,
	"invertscreen":	0,
	"invertfanpolarity":	1,
	"autofanspeed":	0,
	"fanspeed":	100,
	"fanrpm":	5927
}
```

Reply:
Nice, ill implement it.